### PR TITLE
[Add] Skeleton component

### PR DESCRIPTION
**Pull request by nowhhk**
## Description

- it can be customized to any size or style you like.
- By passing a component between Skeleton as children, it will infer its width and height from them.

## Related Issues

I can't solve some type errors

## Tests

I added the following tests:

_Replace this with a list of the tests that you added as part of this PR. A change in behavior with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage](https://codecov.io/gh/dooboolab/dooboo-ui)._

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [ ] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [ ] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: main/Skeleton/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+{
+  loading ? (
+    <Skeleton variant="rect" width={200} height="1.2em"/>
+  ) : (
+    <View>{props.data}</View>
+  );
+}
+```
```
Comment (s-ong-c) [inline main/Skeleton/README.md:42]:
```
       {
          loading 
            ? (
              <Skeleton variant="rect" width={200} height="1.2em"/>
            ) 
            : (
              <View>{props.data}</View>
            )
        }
```
I'd prefer the above code!

File: main/Skeleton/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+<Skeleton
+        variant="circle"
+        width={40}
+        height={40}
+        color="lightgray"
+        animation={false}
+        style={{ marginBottom: 30 }}
+      />
```
Comment (s-ong-c) [inline main/Skeleton/README.md:29]: Please keep your conventation in the Readmd file

File: main/Skeleton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  if (shape === 'circle') {
+    return { borderRadius: 100 };
+  }
+  return undefined;
+};
+
+// Inferring dimensions
+const hasChildren = (children: React.ReactNode): ViewStyle | undefined => {
+  // inline style 있을때
```
Comment (s-ong-c) [inline main/Skeleton/index.tsx:28]: Remove unnecessary comment

File: stories/dooboo-ui/Skeleton.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+      <Row>
+        <Image
+          style={{ height: 100, width: 100 }}
+          source={{
+            uri: 'https://avatars2.githubusercontent.com/u/45788556?s=200&v=4',
+          }}
+        />
+
+        <Skeleton shape="rect">
```
Comment (s-ong-c) [inline stories/dooboo-ui/Skeleton.stories.tsx:99]: Remove unnecessary new line

File: main/Skeleton/index.tsx (hunk 3 above)
Comment (s-ong-c) [reply main/Skeleton/index.tsx:28]: Try not to use Korean since developers around the world will see this.

File: main/Skeleton/index.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+const Skeleton = ({
+  shape,
+  width,
+  height,
+  color,
+  animation,
+  children,
+  style,
+}: Props): ReactElement => {
+  const opacity = new Animated.Value(0.3);
```
Comment (s-ong-c) [inline main/Skeleton/index.tsx:66]:
```
const Skeleton: React.FC<Props> = (props) => {
  const { shape, width, height, color, animation, children, style } = props;
...
```
should be the ordinary typing for functional component.

File: main/Skeleton/index.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+  // inline style 있을때
+  if (children && children.props.style) {
+    return {
+      width: children.props.style.width,
+      height: children.props.style.height,
+    };
+  } else if (children) {
+    return {
+      // width: 'fit-content', height: 'fit-content',
+      width: 'auto', height: 'auto',
```
Comment (s-ong-c) [inline main/Skeleton/index.tsx:37]: `const hasChildren = (children: React.ReactNode): ViewStyle | undefined => {`
is unnecessary
I recommend replacing it with hasChildren with the following code

example
```
customStyle: ViewStyle
defaultStyle: View Style
   style={[children && customStyle ? customStyle : defaultStyle]}>
```

**Comment by JongtaekChoi** [comment]
This PR will be closed due to deactivation. Feel free to reopen this pr if you want to improve it.

**Pull request closed without merging**